libraries/ESP8266WiFiMesh/src/TypeConversionFunctions.cpp

```diff
@@ -56,3 +57,89 @@ uint64_t stringToUint64(const String &string, byte base)
   
   return result;
 }
+
+String uint8ArrayToHexString(const uint8_t *uint8Array, uint32_t arrayLength)
+{
+  char hexString[2*arrayLength + 1]; // Each uint8_t will become two characters (00 to FF) and we want a null terminated char array. 
+  hexString[arrayLength + 1] = { 0 };
+  for(uint32_t i = 0; i < arrayLength; i++)
+  {
+    sprintf(hexString + 2*i, "%02X", uint8Array[i]);
+  }
+
+  return String(hexString);
+}
+
+uint8_t *hexStringToUint8Array(const String &hexString, uint8_t *uint8Array, uint32_t arrayLength)
+{
+  assert(hexString.length() >= arrayLength*2); // Each array element can hold two hexString characters
+  
+  for(uint32_t i = 0; i < arrayLength; i++)
+  {
+    uint8Array[i] = strtoul(hexString.substring(i*2, (i+1)*2).c_str(), nullptr, 16);
+  }
+  
+  return uint8Array;
+}
+
+String macToString(const uint8_t *mac)
+{
+  char macString[13] = { 0 };
+  sprintf(macString, "%02X%02X%02X%02X%02X%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+  return String(macString);
+}
+
+uint8_t *stringToMac(const String &macString, uint8_t *macArray)
+{
+  return hexStringToUint8Array(macString, macArray, 6);
+}
+
+uint64_t macToUint64(const uint8_t *macArray)
```

Comment:
This calculation is expensive. There are some tricks to make 64bit calcs a bit lighter. See the millis internals for one such case.
I don't know if this operation is done in intensive operations, but every bit to lighten the cpu load helps. In particular, this function is "just a conversion function", like a cast, and is easy to forget its weight when using it.
I suggest looking for a shortcut similar to what was done for millis.

Reply:
The updated version of this function is faster by a factor 5 (with the default -Os compiler optimization flag). It also seems to scale from 32 bit to 64 bit linearly, so there is no extra overhead for 64 bit beyond requiring twice the number of operations. I believe this may be due to the fact that it only performs shifts to byte boundaries, which according to the [millis internals](https://github.com/esp8266/Arduino/blob/d9684351c2d11dfe441e89edac43396689ac8ac5/cores/esp8266/core_esp8266_wiring.cpp) you referred to should be "relatively quick".